flexmeasures/api/v2_0/implementations/sensor_data.py

```diff
@@ -0,0 +1,83 @@
+import json
+
+from flask_classful import FlaskView, route
+from flask_security import auth_token_required
+from timely_beliefs import BeliefsDataFrame
+from webargs.flaskparser import use_args
+
+from flexmeasures.api.common.schemas.sensor_data import (
+    GetSensorDataSchema,
+    PostSensorDataSchema,
+)
+from flexmeasures.api.common.utils.api_utils import save_and_enqueue
+from flexmeasures.auth.decorators import account_roles_accepted
+
+
+class SensorDataAPI(FlaskView):
+
+    route_base = "/sensorData"
+    decorators = [auth_token_required]
+
+    @account_roles_accepted("MDC", "Prosumer")
```

Comment:
Yes, we are not handling this at the moment.

To have some auth checking, which also is somewhat making good choices, we should use something like

```
@permission_required_for_context(
        "create-children", arg_loader=AccountIdField.load_current
    )
```

This loads the `Account` of the current user, and checks if the use can create children.
Sensor data is grand-grand-children or an account, but it makes some sense.

Technically (can be done later), we'd want the auth context to be the `Sensor`, but at the moment, I don't know how to easily get it from the sensor data argument. It is available as `bdf.sensor`. If we define arg_loader to be a callable *or* a value (like load_default in marshmallow), we probably can make this work but I haven't tried it:

```
@permission_required_for_context(
        "create-children", arg_loader=bdf.sensor
    )
```

Maybe a ticket for the remaining auth work we'll do later. Or simply a TODO in the code.
But giving this endpoint already some modern auth would be my advice.

Reply:
Alas, I missed your second comment, otherwise I would have notified you that I have been working on this, too.

I have been working on loading the sensor for the context. It turns out one straightforward way to do that is to move the auth decorator into the schema, and use it decorate our `post_load` hook. The reason I think that makes sense is that we shouldn't even need to construct a BeliefsDataFrame (which we do in the post_load hook for the POST endpoint) nor search for data (which we do in the post_load hook for the GET endpoint) in case we already know that the relevant sensor is off limits.

Reply:
Note to others: We discussed offline that our approaches are very similar and I'll go ahead with mine.